OpenRA.Game/Server/Server.cs

```diff
@@ -584,23 +584,28 @@ public void DropClient(Connection toDrop, int frame)
 				// Send disconnected order, even if still in the lobby
 				DispatchOrdersToClients(toDrop, 0, new ServerOrder("Disconnected", "").Serialize());
 
-				LobbyInfo.Clients.RemoveAll(c => c.Index == toDrop.PlayerIndex);
-
-				// Client was the server admin
-				// TODO: Reassign admin for game in progress via an order
-				if (Dedicated && dropClient.IsAdmin && State == ServerState.WaitingPlayers)
+				// Client was the server admin, but the game isn't over yet.
+				if (Dedicated && dropClient.IsAdmin)
 				{
-					// Remove any bots controlled by the admin
-					LobbyInfo.Clients.RemoveAll(c => c.Bot != null && c.BotControllerClientIndex == toDrop.PlayerIndex);
+					// Still in the lobby so handle things directly.
+					if (State == ServerState.WaitingPlayers)
+					{
+						// Remove the previous admin.
+						LobbyInfo.Clients.RemoveAll(c => c.Index == toDrop.PlayerIndex);
 
-					var nextAdmin = LobbyInfo.Clients.Where(c1 => c1.Bot == null)
-						.MinByOrDefault(c => c.Index);
+						// Remove any bots controlled by the previous admin.
+						LobbyInfo.Clients.RemoveAll(c => c.Bot != null && c.BotControllerClientIndex == toDrop.PlayerIndex);
 
-					if (nextAdmin != null)
-					{
-						nextAdmin.IsAdmin = true;
-						SendMessage("{0} is now the admin.".F(nextAdmin.Name));
+						// Find a new admin.
+						var nextAdmin = LobbyInfo.Clients.Where(c1 => c1.Bot == null).MinByOrDefault(c => c.Index);
+						if (nextAdmin != null)
+						{
+							nextAdmin.IsAdmin = true;
+							SendMessage("{0} is now the admin.".F(nextAdmin.Name));
+						}
 					}
+					else
+						DispatchOrdersToClients(toDrop, 0, new ServerOrder("NewAdmin", "").Serialize());
```

Comment:
Both
```csharp
DispatchOrders(toDrop, toDrop.MostRecentFrame, Order.NewAdmin(nextAdmin.Name).Serialize());
```

```csharp
DispatchOrdersToClients(toDrop, toDrop.MostRecentFrame, Order.NewAdmin(nextAdmin.Name).Serialize());
```

cause
```
Exception of type `System.ArgumentException`: An item with the same key has already been added.
  at System.ThrowHelper.ThrowArgumentException (System.ExceptionResource resource) [0x0000b] in <8f2c484307284b51944a1a13a14c0266>:0 
  at System.Collections.Generic.Dictionary`2[TKey,TValue].Insert (TKey key, TValue value, System.Boolean add) [0x0008e] in <8f2c484307284b51944a1a13a14c0266>:0 
  at System.Collections.Generic.Dictionary`2[TKey,TValue].Add (TKey key, TValue value) [0x00000] in <8f2c484307284b51944a1a13a14c0266>:0 
  at OpenRA.Network.FrameData.AddFrameOrders (System.Int32 clientId, System.Int32 frame, System.Byte[] orders) [0x0000e] in <67c5022a3d434b6a913576734b344516>:0 
  at OpenRA.Network.OrderManager+<TickImmediate>c__AnonStorey0.<>m__0 (System.Int32 clientId, System.Byte[] packet) [0x00077] in <67c5022a3d434b6a913576734b344516>:0 
  at OpenRA.Network.EchoConnection.Receive (System.Action`2[T1,T2] packetFn) [0x00047] in <67c5022a3d434b6a913576734b344516>:0 
  at OpenRA.Network.OrderManager.TickImmediate () [0x000b4] in <67c5022a3d434b6a913576734b344516>:0 
  at OpenRA.Sync+<CheckSyncUnchanged>c__AnonStorey0.<>m__0 () [0x00001] in <67c5022a3d434b6a913576734b344516>:0 
  at OpenRA.Sync.CheckSyncUnchanged[T] (OpenRA.World world, System.Func`1[TResult] fn) [0x00043] in <67c5022a3d434b6a913576734b344516>:0 
  at OpenRA.Sync.CheckSyncUnchanged (OpenRA.World world, System.Action fn) [0x0000e] in <67c5022a3d434b6a913576734b344516>:0 
  at OpenRA.Game.InnerLogicTick (OpenRA.Network.OrderManager orderManager) [0x0011a] in <67c5022a3d434b6a913576734b344516>:0 
  at OpenRA.Game.LogicTick () [0x00050] in <67c5022a3d434b6a913576734b344516>:0 
  at OpenRA.Game.Loop () [0x000d6] in <67c5022a3d434b6a913576734b344516>:0 
  at OpenRA.Game.Run () [0x00042] in <67c5022a3d434b6a913576734b344516>:0 
  at OpenRA.Program.Run (System.String[] args) [0x00011] in <67c5022a3d434b6a913576734b344516>:0 
  at OpenRA.Program.Main (System.String[] args) [0x0004f] in <67c5022a3d434b6a913576734b344516>:0 
```

on the remaining clients. This doesn't happen when dispatching with frame `0`.